controller/src/controller/models/domain/rsptag.py

```diff
@@ -349,6 +368,55 @@ def _determine_build(
         else:
             return rest if rest else None
 
+    @staticmethod
+    def _calculate_date(tagdata: dict[str, str]) -> datetime.datetime | None:
+        """Calculate the date when the image should have been created.
+
+        Parameters
+        ----------
+        tagdata
+            The match groups from the regular expression tag match.
+
+        Returns
+        -------
+        datetime.datetime | None
+            The image creation date if it can be gleaned from the tag.
+        """
+        year = tagdata.get("year")
+        if not year:
+            return None
+        week = tagdata.get("week")
+        if year and week:
+            thursday = 4  # We build on Thursday, and Monday is index 1.
+            stamp = datetime.datetime.fromisocalendar(
+                int(year), int(week), thursday
+            )
+            # fromisocalendar() gives us a naive object; force it to UTC
+            return stamp.replace(tzinfo=datetime.UTC)
+        month = tagdata.get("month")
+        day = tagdata.get("day")
+        if not (month and day):
+            return None
+        return datetime.datetime(
+            int(year), int(month), int(day), tzinfo=datetime.UTC
+        )
+
+    def age(
+        self, since: datetime.datetime | None = None
+    ) -> datetime.timedelta | None:
+        """Calculate image age, if possible.
+
+        Parameters
+        ----------
+        since
+            Datestamp to measure age from.  If unspecified, the present.
+        """
+        if self.date is None:
+            return None
+        if since is None:
+            since = datetime.datetime.now(datetime.UTC)
```

Comment:
```suggestion
            since = datetime.now(tz=UTC)
```
I guess the `tz=` technically isn't needed but that's what Ruff rewrites things to, so it's what we've been doing most other places.